zucchini-ui-frontend/src/stepDefinitions/components/StepDefinitionsTable.jsx

```diff
@@ -0,0 +1,255 @@
+    text: PropTypes.string.isRequired,
+    styleClass: PropTypes.string.isRequired,
+  };
+
+  render() {
+    const { text, styleClass } = this.props;
+    return (
+      <span className={styleClass}>{text}</span>
+    );
+  }
+}
+
+class StepDefinitionsVariantsDialog extends React.PureComponent {
+
+  static propTypes = {
+    show: PropTypes.bool.isRequired,
+    location: PropTypes.string.isRequired,
+    occurrences: PropTypes.arrayOf(PropTypes.object).isRequired,
+    onClose: PropTypes.func.isRequired,
+  };
+
+  onCloseClick = (event) => {
+    if (event) {
+      event.preventDefault();
+    }
+    this.props.onClose();
+  };
+
+  render() {
+    const { show, occurrences, location } = this.props;
+    return (
+      <Modal bsSize='large' dialogClassName='modal-dialog' show={show} onHide={this.onCloseClick}>
+        <Modal.Header closeButton>
+          <Modal.Title>{'Variantes Connues'} <span className='step-definition-location'>{location}</span></Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <pre>{formatStepDefinitionVariants(occurrences)}</pre>
+        </Modal.Body>
+      </Modal>
+    );
+  }
+}
+
+function groupDistinctArgumentsByOffset(occurrences) {
+  const groupedArgs = {};
+  const knownOffsets = occurrences[0].info.arguments.map(arg => {
+    return arg.offset
+  });
+
+  if (knownOffsets.length > 0) {
+    occurrences.forEach(occurrence => {
+      occurrence.info.arguments.map(argument => {
+        const key = getClosestOffset(knownOffsets, argument.offset);
+        const value = argument.value;
+        if (groupedArgs[key] === undefined) {
+          groupedArgs[key] = [value];
+        } else {
+          // No value duplicates
+          if (!includes(groupedArgs[key], value)) {
+            groupedArgs[key].push(value);
+          }
+        }
+      })
+    });
+  }
+  return groupedArgs;
+}
+
+function getClosestOffset(knownOffsets, currentOffset) {
+  // Sometimes the offset returned by cucumber is slightly shifted due to non-capturing groups in regexes
+  // Ex: " this is a glue with non capturing groups?" -> The "s" in "groups" will shift ethe offset by one
+  if (includes(knownOffsets[currentOffset])) {
+    return currentOffset;
+  }
+  return knownOffsets.reduce((prev, curr) => Math.abs(curr - currentOffset) < Math.abs(prev - currentOffset) ? curr : prev);
+}
+
+function formatVariantsBlock(variants, variantStyle) {
+  return variants.map(variant => {
+    return <StepDefinitionsTableHighlightedTerm key={variant} text={variant} styleClass={variantStyle} />
+  });
+}
+
+function formatStepDefinition(stepDefinition) {
+
+  const keyword = stepDefinition.info.keyword;
+  const definition = truncate(stepDefinition.info.name, { 'length': 300 });
+  const args = stepDefinition.info.arguments;
+
+  const formattedText = [];
+  formattedText.push(<span key={Math.random()}><StepDefinitionsTableHighlightedTerm text={keyword} styleClass='step-definition-keyword' />&nbsp;</span>);
```

Comment:
Never use a random for the key attribute in React. Use an index variable for all keys.